src/amuse/ext/orbital_elements.py

```diff
@@ -27,81 +28,228 @@ def newton(f,x0,fprime=None,args=(),tol=1.48e-8,maxiter=50):
+            (
+                vectors
+                * secondary_mass.reshape(
+                    (len(secondary_mass), 1)
+                    )
+                )
+            / total_mass
+            )
+    return center_of_mass_array
+
+
+def generate_binaries_from_orbital_elements(
+        primary_mass,
+        secondary_mass,
+        semi_major_axis,
+        eccentricity=0 | units.rad,
+        true_anomaly=0 | units.rad,
+        inclination=0 | units.rad,
+        longitude_of_the_ascending_node=0 | units.rad,
+        argument_of_periapsis=0 | units.rad,
+        G=nbody_system.G
+        ):
+    """
+    returns two particlesets, which contain the primaries and the secondaries
+    in binary pairs.
+    """
+    number_of_primaries = len(primary_mass)
+    number_of_secondaries = len(secondary_mass)
+
+    # mass arrays need to be the same length
+    if number_of_secondaries != number_of_primaries:
+        raise Exception("The number of primaries is not the same as the number\
+                of secondaries, this is not supported.")
+    # other arrays need to be the same length as well, or have just one value
+    semi_major_axis = equal_length_array_or_scalar(
+            semi_major_axis, length=number_of_primaries)
+    eccentricity = equal_length_array_or_scalar(
+            eccentricity, length=number_of_primaries)
+    true_anomaly = equal_length_array_or_scalar(
+            true_anomaly, length=number_of_primaries)
+    inclination = equal_length_array_or_scalar(
+            inclination, length=number_of_primaries)
+    longitude_of_the_ascending_node = equal_length_array_or_scalar(
+            longitude_of_the_ascending_node, length=number_of_primaries)
+    argument_of_periapsis = equal_length_array_or_scalar(
+            argument_of_periapsis, length=number_of_primaries)
+
+    cos_true_anomaly = cos(true_anomaly)
+    sin_true_anomaly = sin(true_anomaly)
+
+    cos_inclination = cos(inclination)
+    sin_inclination = sin(inclination)
+
+    cos_arg_per = cos(argument_of_periapsis)
+    sin_arg_per = sin(argument_of_periapsis)
+
+    cos_long_asc_nodes = cos(longitude_of_the_ascending_node)
+    sin_long_asc_nodes = sin(longitude_of_the_ascending_node)
+
+    # alpha is a unit vector directed along the line of node
+    alphax = (
+            cos_long_asc_nodes*cos_arg_per
+            - sin_long_asc_nodes*sin_arg_per*cos_inclination
+            )
+    alphay = (
+            sin_long_asc_nodes*cos_arg_per
+            + cos_long_asc_nodes*sin_arg_per*cos_inclination
+            )
     alphaz = sin_arg_per*sin_inclination
-    alpha = numpy.array([alphax,alphay,alphaz])
-
-    ### beta is a unit vector perpendicular to alpha and the orbital angular momentum vector ###
-    betax = -cos_long_asc_nodes*sin_arg_per - sin_long_asc_nodes*cos_arg_per*cos_inclination
-    betay = -sin_long_asc_nodes*sin_arg_per + cos_long_asc_nodes*cos_arg_per*cos_inclination
+    alpha = numpy.array([alphax, alphay, alphaz])
```

Comment:
one thing to watch for is whether numpy arrays of python objects (ie quantities) are generated (this will slow down things tremendously)  - this may do that  (but maybe not)